added atoms-molecule-organism storybook

EmanFouad opened the pull request:
**Description**

**Added Atoms for:**
  * Avatars ( Large , Medium , Small , Filled , Anonymous , Not Filled )
  * Button ( Primary , Primary+Icon , Primary+Icon Only )
  * Checkboxes
  * Dividers
  * Image Upload
   * Inputs ( Choices , Primary Label , Primary Label+Icon )
  * Toggles
  * Radios

**Added Molecule for :**
 * Tab

**Added Static Organism for:**
 * Image Poll - Default

EmanFouad committed:
added atoms-molecules-organisms

EmanFouad committed:
conflict solve

amagdy46 commented:
- this is not a review. but in future try to spread your code over multiple relatable commits.

Sahl-A commented on storybook/src/Eman/Atoms/Avatars/Avatar.js:
```diff
@@ -0,0 +1,87 @@
+};
+
+Avater.propTypes = {
+  size: PropTypes.string,
```
You can use PropTypes.oneOf(['first value','Second Value','Third Value'])
This way you ensure no other than these values are passed to the component

Sahl-A commented on storybook/src/Eman/Atoms/Avatars/Avatar.js:
```diff
@@ -0,0 +1,87 @@
+
+Avater.propTypes = {
+  size: PropTypes.string,
+  type: PropTypes.string,
```
Same as line above

Sahl-A commented:
Great Job Eman. 
Since there are many changes, atoms, molecules & organisms added, I will review it By component:

-Avatar
-- Great use of storybook controls
-- I guess it is better if you import the SVGs instead of writing them directly. This makes the code more readable
-- Can you add a way to add the image when choosing type="filled"

-Buttons
-- In Primary Icon Only & Primary Icon buttons, There are some controls without any names, I can't determine what there functions are
--Secondary Variant is activated only in the 3rd button, The others stuck with the primary
-- Disabled works only in the 3rd button

Sahl-A commented on storybook/src/Eman/Atoms/Checkboxes/Checkbox.js:
```diff
@@ -0,0 +1,60 @@
+  return (
+    <>
+      {size === CHECKBOX.SIZE.DEFAULT && status && (
+        <button className="w-6 h-6 rounded-sm bg-primary flex justify-center  items-center">
```
for semantic and accessibility reasons, it is better use input with type=checkbox instead of button.
This applies to all checkbox implementation below

Sahl-A commented on storybook/src/Eman/Atoms/Checkboxes/Checkbox.js:
```diff
@@ -0,0 +1,60 @@
+
+Checkbox.propTypes = {
+  status: PropTypes.bool,
+  size: PropTypes.string,
```
size: PropTypes.oneOf([ 'value1', 'value2', 'value3']) is better as it is implemented in react not storybook.
Also if you use it, you can it will be used by storybook directly without passing the props in the stories file.

Sahl-A commented on storybook/src/Eman/Atoms/Radios/Radio.js:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <>
+      {size === RADIO.SIZE.DEFAULT && status === RADIO.STATUS.ON && (
+        <button className="w-6 h-6 rounded-lg bg-primary flex justify-center items-center hover:bg-primary-shd8 focus:border-2 focus:border-primary-shd6 focus:outline-none">
```
Use input type=radio instead of button

Sahl-A commented on storybook/src/Eman/Atoms/Toggles/Toggle.js:
```diff
@@ -0,0 +1,45 @@
+  return (
+    <>
+      {size === TOGGLE.SIZE.DEFAULT && status === TOGGLE.STATUS.ON && (
+        <button className="bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border w-10 h-6 rounded-full flex items-center flex-row-reverse">
```
Use  ` <input type=Checkbox/>` instead of button

Sahl-A commented:
- CheckBox
-- There is no disable state 
-- The checkbox is not clickable. It does not check/uncheck when clicking it
- Divider
-- It would be better if you add a control to change its color and increase its range. This is an opinion not a comment though :D 
- Inputs
-- It has the same buttons' issue with controls. You should group them separately, You have 3 different components not 3 different story for one component
-Choices:-
--- Can't control the choice in the component
-Primary input;-
--- Not all the input field is clickable. You must click it from left side
-Primary label Icon:- 
--- Can't change the home icon. It is not dynamic
- Radio
-- It is not clickable
- Toggle
-- It is not clickable
######################################
Molecules
########
- Tabs
-- Hover is not a state we pass as a prop, better to implement it 
-- Radio can still be on/off while statues is disabled
-- The Tab is not clickable neither the Radio button inside it
#######################################
Organisms
##########
- Image Poll
-- It is just a UI representation to the image Poll, it is not dynamic/reusable neither any of its content except for the upload button
If you intended to do it so, you can point this out in description.

Sahl-A reviewed (changes requested):
Please consider the general comments also

EmanFouad committed:
edit story title and propTypes

EmanFouad committed:
made the required changes

EmanFouad commented:
**Changes**
-----------------------------------------------------------------------------------------------
**Avatar**
* added imgName Field to change images using static path
* you should use one of this names to change images ( img2.e3c2ff2f.jfif , img3.c3bd2696.jpg , img.dd659847.jfif )

**Checkbox**
* clickable checkbox has been implemented
* added ( hover , focus , disabled ) effects

**Divider**
* added a control to change its background color

**Choices**
* basic function has been implemented to generate dynamic letters

**Inputs**
* regarding this ( Primary label Icon : Can't change the home icon. It is not dynamic ) , I tried to make it dynamic but i couldn't :(

**Radio & Toggle**
* clickable radio has been implemented
* added ( disabled ) effect

**Tabs**
* Hover has been implemented as a style
*  clickable tab has been implemented

**Organism**
* Edited its description

**General Notes :**
* solved the Controls issue
* refactor previous implementation for most of components to make code more readable
* made each svg as independent component

Sahl-A reviewed (dismissed):
Thank you Eman for this thorough changes. Most of the atoms have been changed and refactored. Functionality also has been added. The below comments to consider later:

- You made SVGs a reusable component. However, consider consistency in its props. Some SVGs have 1 prop, while others have 2. Making all of them receive the same props would make your SVG more reusable and predictable. 

- The tab is clickable yes, however, the radio button shouldn't be Switched on/off by click. The radio button should be switched only if another radio button with same name attr is chosen.

EmanFouad committed:
Delete .eslintcache
